src/scripts/generate-unary-ops.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
```

Comment:
In general I can suggest to use classes, member functions, and inheritances instead of lots of functions in a python file.
In addition the binary and the unary operators evaluation are similar. So, you can reuse some of the logic.